### DISCUSSION TITLE
ValueError: coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data

### DISCUSSION THREAD

**Discussion by Constable1819** [pypsa-meets-earth/pypsa-earth#1124] [General]
I have started the process from the beginning. I got this error for the tutorial that was even running before. 

ValueError: coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data

**Comment by ekatef** [comment]
> I have started the process from the beginning. I got this error for the tutorial that was even running before.
> 
> ValueError: coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data

Hello @Constable1819, thanks for the report. Could you please post the whole error traceback?

Just to be sure: have you removed all the previous output of the workflow before running the tutorial? (It sounds very trivial but interactions with the old output are a quite frequent source of errors

**Reply by Constable1819** [reply]
ValueError: coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data
[Mon Sep 30 22:58:41 2024]
Error in rule build_renewable_profiles:
    jobid: 18
    input: resources/natura.tiff, data/copernicus/PROBAV_LC100_global_v3.0.1_2019-nrt_Discrete-Classification-map_EPSG-4326.tif, data/gebco/GEBCO_2021_TID.nc, resources/shapes/country_shapes.geojson, resources/shapes/offshore_shapes.geojson, data/hydro_capacities.csv, data/eia_hydro_annual_generation.csv, resources/powerplants.csv, resources/bus_regions/regions_onshore.geojson, cutouts/cutout-2013-era5-tutorial.nc
    output: resources/renewable_profiles/profile_solar.nc
    log: logs/build_renewable_profile_solar.log (check log file(s) for error details)

RuleException:
CalledProcessError in file /Users/xxx/pypsa-earth/Snakefile, line 484:
Command 'set -euo pipefail;  /opt/miniconda3/envs/pypsa-earth/bin/python3.10 /Users/xxx/pypsa-earth/.snakemake/scripts/tmp5_ny9xee.build_renewable_profiles.py' returned non-zero exit status 1.
  File "/Users/xxx/pypsa-earth/Snakefile", line 484, in __rule_build_renewable_profiles
  File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-09-30T225500.825267.snakemake.log

**Reply by ekatef** [reply]
> ValueError: coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data [Mon Sep 30 22:58:41 2024] Error in rule build_renewable_profiles: jobid: 18 input: resources/natura.tiff, data/copernicus/PROBAV_LC100_global_v3.0.1_2019-nrt_Discrete-Classification-map_EPSG-4326.tif, data/gebco/GEBCO_2021_TID.nc, resources/shapes/country_shapes.geojson, resources/shapes/offshore_shapes.geojson, data/hydro_capacities.csv, data/eia_hydro_annual_generation.csv, resources/powerplants.csv, resources/bus_regions/regions_onshore.geojson, cutouts/cutout-2013-era5-tutorial.nc output: resources/renewable_profiles/profile_solar.nc log: logs/build_renewable_profile_solar.log (check log file(s) for error details)
> 
> RuleException: CalledProcessError in file /Users/xxx/pypsa-earth/Snakefile, line 484: Command 'set -euo pipefail; /opt/miniconda3/envs/pypsa-earth/bin/python3.10 /Users/xxx/pypsa-earth/.snakemake/scripts/tmp5_ny9xee.build_renewable_profiles.py' returned non-zero exit status 1. File "/Users/xxx/pypsa-earth/Snakefile", line 484, in __rule_build_renewable_profiles File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/concurrent/futures/thread.py", line 58, in run Shutting down, this might take some time. Exiting because a job execution failed. Look above for error message Complete log: .snakemake/log/2024-09-30T225500.825267.snakemake.log

Thank you! Is it all the traceback you got? It would be great to see in which particular fragment of the code the error happens.

From the errors messages, I can guess that something is wrong with the cutout, like it doesn't cover the requested area or snapshots, but it would be great to have more details

**Reply by Constable1819** [reply]
[Mon Sep 30 22:58:19 2024]
rule build_renewable_profiles:
    input: resources/natura.tiff, data/copernicus/PROBAV_LC100_global_v3.0.1_2019-nrt_Discrete-Classification-map_EPSG-4326.tif, data/gebco/GEBCO_2021_TID.nc, resources/shapes/country_shapes.geojson, resources/shapes/offshore_shapes.geojson, data/hydro_capacities.csv, data/eia_hydro_annual_generation.csv, resources/powerplants.csv, resources/bus_regions/regions_onshore.geojson, cutouts/cutout-2013-era5-tutorial.nc
    output: resources/renewable_profiles/profile_solar.nc
    log: logs/build_renewable_profile_solar.log
    jobid: 18
    benchmark: benchmarks/build_renewable_profiles_solar
    reason: Missing output files: resources/renewable_profiles/profile_solar.nc; Input files updated by another job: cutouts/cutout-2013-era5-tutorial.nc, data/copernicus/PROBAV_LC100_global_v3.0.1_2019-nrt_Discrete-Classification-map_EPSG-4326.tif, data/gebco/GEBCO_2021_TID.nc, resources/shapes/offshore_shapes.geojson, resources/powerplants.csv, resources/bus_regions/regions_onshore.geojson, resources/shapes/country_shapes.geojson, resources/natura.tiff
    wildcards: technology=solar
    resources: tmpdir=/var/folders/r9/7fg7nyxj4m3_cb47j_qgsqw80000gn/T, mem_mb=20000, mem_mib=19074

/opt/miniconda3/envs/pypsa-earth/lib/python3.10/site-packages/pypsa/networkclustering.py:16: UserWarning: The namespace `pypsa.networkclustering` is deprecated and will be removed in PyPSA v0.24. Please use `pypsa.clustering.spatial instead`. 
  warnings.warn(
correction_factor is set as 0.854337
INFO:__main__:correction_factor is set as 0.854337
Calculate landuse availabilities...
INFO:__main__:Calculate landuse availabilities...
/opt/miniconda3/envs/pypsa-earth/lib/python3.10/site-packages/pypsa/networkclustering.py:16: UserWarning: The namespace `pypsa.networkclustering` is deprecated and will be removed in PyPSA v0.24. Please use `pypsa.clustering.spatial instead`. 
  warnings.warn(
Traceback (most recent call last):
  File "/Users/xxx/pypsa-earth/.snakemake/scripts/tmp5_ny9xee.build_renewable_profiles.py", line 741, in <module>
    availability = cutout.availabilitymatrix(regions, excluder, **kwargs)
  File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/site-packages/atlite/gis.py", line 761, in compute_availabilitymatrix
    return xr.DataArray(availability, coords=coords)
  File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/site-packages/xarray/core/dataarray.py", line 445, in __init__
    coords, dims = _infer_coords_and_dims(data.shape, coords, dims)
  File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/site-packages/xarray/core/dataarray.py", line 147, in _infer_coords_and_dims
    raise ValueError(
ValueError: coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data
[Mon Sep 30 22:58:41 2024]
Error in rule build_renewable_profiles:
    jobid: 18
    input: resources/natura.tiff, data/copernicus/PROBAV_LC100_global_v3.0.1_2019-nrt_Discrete-Classification-map_EPSG-4326.tif, data/gebco/GEBCO_2021_TID.nc, resources/shapes/country_shapes.geojson, resources/shapes/offshore_shapes.geojson, data/hydro_capacities.csv, data/eia_hydro_annual_generation.csv, resources/powerplants.csv, resources/bus_regions/regions_onshore.geojson, cutouts/cutout-2013-era5-tutorial.nc
    output: resources/renewable_profiles/profile_solar.nc
    log: logs/build_renewable_profile_solar.log (check log file(s) for error details)

RuleException:
CalledProcessError in file /Users/xxx/pypsa-earth/Snakefile, line 484:
Command 'set -euo pipefail;  /opt/miniconda3/envs/pypsa-earth/bin/python3.10 /Users/xxx/pypsa-earth/.snakemake/scripts/tmp5_ny9xee.build_renewable_profiles.py' returned non-zero exit status 1.
  File "/Users/xxx/pypsa-earth/Snakefile", line 484, in __rule_build_renewable_profiles
  File "/opt/miniconda3/envs/pypsa-earth/lib/python3.10/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-09-30T225500.825267.snakemake.log

**Comment by ekatef** [comment]
@Constable1819 thank you!! It appears that the reason is the mismatch between the requested countries and the cutout area and/or `natura.tiff` raster. Coild you please let know what is the list of countries you currently define in `config.yaml`?

**Reply by Constable1819** [reply]
@ekatef this is the tutorial config ["NG,BJ"], no edits have been made

**Comment by futuredataperson** [comment]
Halo  @Constable1819  @ekatef I am getting a similar error for the tutorial too;  coords is not dict-like, but it has 3 items, which does not match the 4 dimensions of the data. Have you found a solution?

**Reply by ekatef** [reply]
Hello @Constable1819 @futuredataperson, thanks for reporting the troubles. According [the discussion](https://discord.com/channels/911692131440148490/1289961191694929952) in Support channel of PyPSA-meets-Earth server, the reason of these troubles is an update of `rasterio` version. Have created #1146 PR to fix that. Could you please test if using [this](https://github.com/ekatef/pypsa-earth/blob/env_fix/envs/environment.yaml )`environment.yaml` resolve the issue?

**Reply by Constable1819** [reply]
Hi @ekatef Yes it does. Thank you.

**Reply by ekatef** [reply]
> Hi @ekatef Yes it does. Thank you.

Super, thank you so much for the confirmation
